src/inmanta/logging.py

```diff
@@ -423,7 +690,107 @@ def format(self, record: logging.LogRecord) -> str:
         :param record: The `logging.LogRecord` object to format.
         :return: The formatted log record as a string.
         """
-        record = self._logger_config.wrap_record(record)
+        record = self._wrap_record(record)
         indent: str = " " * self.get_header_length(record)
         head, *tail = super().format(record).splitlines(True)
         return head + "".join(indent + line for line in tail)
+
+    def _wrap_record(self, record: logging.LogRecord) -> logging.LogRecord:
+        """
+        Wrap a log record to perform renaming for specific formatter as determined by the _logger_mode
+
+        This is derived from the way the colorlog.ColoredFormatter works
+        """
+        old_name = record.name
+        new_name = self._get_logger_name_for(old_name)
+        if old_name == new_name:
+            return record
+        attributes = dict(record.__dict__)
+        attributes["name"] = new_name
+        return logging.makeLogRecord(attributes)
+
+    def _get_logger_name_for(self, logger_name: str) -> str:
+        """
+        Returns the logger name that should be used in the log record.
+
+        :attr logger_name: The name of the logger that was used to create the log record.
+        """
+        logger_mode = self._logger_mode_manager.get_logger_mode()
+        if not self._keep_logger_names and logger_mode in [LoggerMode.COMPILER, LoggerMode.EXPORTER]:
+            if not logger_name.startswith("inmanta"):
+                # This is a log record from a third-party library. Don't adjust the logger name.
+                return logger_name
+            if logger_name == "inmanta.pip":
+                # Log record created by a pip subprocess started by the inmanta.
+                return "pip"
+            match: Optional[re.Match[str]] = self.inmanta_plugin_pkg_regex.match(logger_name)
+            if match:
+                # Log record created by an Inmanta module.
+                return match.groupdict()["module_name"]
+            else:
+                # Log record created by Inmanta code.
+                return logger_mode.value
+        else:
+            # Don't modify the logger name
+            return logger_name
+
+
+class MultiFileHandler(logging.Handler):
+    """
+    A file handler that writes to different files based on the last part of the logger name.
+    """
+
+    def __init__(self, name_parent_logger: str, log_file_template: str) -> None:
+        """
+        :param name_parent_logger: The log records created by children of this logger will be written to file by this handler.
+                                   This handler will ignore log records created by any other logger.
+        :param log_file_template: A template for the path to the log file. This is an f-string that holds the parameter
```

Comment:
Currently, we parametrize on the full child logger name. It might be a bit more stable to use only the first level below the parent, so you wouldn't get `/path/to/dir/<env_id>.sublogger.log`.

Then again, afaik we currently don't ever create subloggers for the resource action log.